vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java

```diff
@@ -4090,10 +4099,61 @@ public void setNestedNullBehavior(NestedNullBehavior nestedNullBehavior) {
 
     /**
      * Get the behavior when facing nested <code>null</code> values.
-     * 
+     *
      * @return The current behavior when facing nested <code>null</code> values.
      */
     public NestedNullBehavior getNestedNullBehavior() {
     	return nestedNullBehavior;
     }
+
+    private void onInMemoryFilterOrSortingChange(
+            SerializablePredicate<T> filter,
+            SerializableComparator<T> sortComparator) {
+        updateSorting(sortComparator);
+        updateInMemoryFiltering(filter);
+
+        dataCommunicator.reset();
+    }
+
+    private void updateInMemoryFiltering(
+            SerializablePredicate<T> componentInMemoryFilter) {
+        assert filterSlot != null : "Filter Slot is supposed not to be empty when set the filter";
+        // As long as the Grid currently contains only in-memory filter
+        // and only list data view has a filter setup API, we can safely cast
+        // the filter slot type into in-memory filter (predicate).
+        @SuppressWarnings("unchecked")
+        SerializableConsumer<SerializablePredicate<T>> inMemoryFilter = (SerializableConsumer<SerializablePredicate<T>>) filterSlot;
+        inMemoryFilter
+                .accept(componentInMemoryFilter);
+    }
+
+    private void updateSorting(SerializableComparator<T> componentSorting) {
```

Comment:
Adding yet another internal sorting method on in addition to
`setSortOrder(List<GridSortOrder<T>> order, boolean userOriginated)`
and `private void sort(boolean userOriginated)` and it makes the code seriously less maintainable.

Since you probably know what you're doing and are on top of the relation between these three methods, can you please add some javadocs that explain what case(?) the method is actually for as the names are very interchangeable. It would help keep the code more maintainable

Reply:
Javadocs added.